classes/log/EmailLogEntry.php

```diff
@@ -127,39 +126,29 @@ public function setAssocId($assocId)
     }
 
     /**
-     * Return the full name of the sender (not necessarily the same as the from address).
-     *
-     * @return string
+     * Return the sender user.
      */
-    public function getSenderFullName()
+    public function getSender(): ?User
     {
-        $senderFullName = $this->getData('senderFullName');
-
-        if ($senderFullName) {
-            return $senderFullName;
-        }
-
-        $sender = $this->getSenderId()
-            ? Repo::user()->get($this->getSenderId(), true)
-            : null;
+        return $this->sender ??= $this->getSenderId() ? Repo::user()->get($this->getSenderId(), true)?->getFullName() : null;
+    }
 
-        return $sender ? $sender->getFullName() : '';
+    /**
+     * Return the full name of the sender (not necessarily the same as the from address).
+     */
+    public function getSenderFullName(): string
```

Comment:
In cases like this, I'd rather permit a `null` return to indicate that the data is missing. That means data presence/absence stays type-based (`string` vs `null`) rather than value-based (`'something'` vs `''`), which is conceptually more clear.

If it should always be the case that a value is returned, then having this function throw a `TypeError` or similar is probably not a bad thing.

Reply:
In general, I prefer to avoid nulls, unless it's really needed (price: 0 vs null). Otherwise we need to be careful when passing it around (e.g. `strlen($name ?? '')`).